-1

Tested at: 95e72d226b7812f94bd1f284887e91a4d9f8e261

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-15385/14418/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  ClangBuild**

* **Build**:

I found an error when building: 
 
 <pre>>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-08-08-1100/src/Alignment/MillePedeAlignmentAlgorithm/plugins/MillePedeFileConverter.cc 
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-08-08-1100/src/Utilities/StaticAnalyzers/src/EDMPluginDumper.cc 
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-08-08-1100/src/Utilities/StaticAnalyzers/src/FiniteMathChecker.cc 
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-08-08-1100/src/Utilities/StaticAnalyzers/src/UsingNamespace.cpp 
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-08-08-1100/src/Alignment/MillePedeAlignmentAlgorithm/plugins/MillePedeDQMModule.cc: In member function 'void MillePedeDQMModule::fillExpertHistos()':
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-08-08-1100/src/Alignment/MillePedeAlignmentAlgorithm/plugins/MillePedeDQMModule.cc:84:80: error: 'class MillePedeFileReader' has no member named 'getXobs'
   fillExpertHisto(h_xPos,  Xcut_, sigCut_, maxMoveCut_, maxErrorCut_, mpReader.getXobs(),  mpReader.getXobsErr());
                                                                                ^
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-08-08-1100/src/Alignment/MillePedeAlignmentAlgorithm/plugins/MillePedeDQMModule.cc:84:101: error: 'class MillePedeFileReader' has no member named 'getXobsErr'
   fillExpertHisto(h_xPos,  Xcut_, sigCut_, maxMoveCut_, maxErrorCut_, mpReader.getXobs(),  mpReader.getXobsErr());
                                                                                                     ^
</pre>

* **Clang**:

I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && scram build -k -j 16 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'


 <pre>>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-08-08-1100/src/Alignment/MillePedeAlignmentAlgorithm/plugins/PedeLabeler.cc 
>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-08-08-1100/src/Alignment/MillePedeAlignmentAlgorithm/plugins/MillePedeAlignmentAlgorithm.cc 
>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-08-08-1100/src/Alignment/MillePedeAlignmentAlgorithm/plugins/MillePedeFileExtractor.cc 
>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-08-08-1100/src/Alignment/MillePedeAlignmentAlgorithm/plugins/MillePedeFileConverter.cc 
>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-08-08-1100/src/Alignment/MillePedeAlignmentAlgorithm/plugins/MomentumDependentPedeLabeler.cc 
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-08-08-1100/src/Alignment/MillePedeAlignmentAlgorithm/plugins/MillePedeDQMModule.cc:84:80: error: no member named 'getXobs' in 'MillePedeFileReader'
  fillExpertHisto(h_xPos,  Xcut_, sigCut_, maxMoveCut_, maxErrorCut_, mpReader.getXobs(),  mpReader.getXobsErr());
                                                                      ~~~~~~~~ ^
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-08-08-1100/src/Alignment/MillePedeAlignmentAlgorithm/plugins/MillePedeDQMModule.cc:84:101: error: no member named 'getXobsErr' in 'MillePedeFileReader'
  fillExpertHisto(h_xPos,  Xcut_, sigCut_, maxMoveCut_, maxErrorCut_, mpReader.getXobs(),  mpReader.getXobsErr());
                                                                                           ~~~~~~~~ ^
</pre>

--- a/Alignment/MillePedeAlignmentAlgorithm/interface/MillePedeFileReader.h
+++ b/Alignment/MillePedeAlignmentAlgorithm/interface/MillePedeFileReader.h
@@ -20,6 +20,21 @@ class MillePedeFileReader {
     void read();
     bool storeAlignments();
 
+    std::array<double, 6> const& getXobs()     const { return Xobs;     }
+    std::array<double, 6> const& getXobsErr()  const { return XobsErr;  }
+    std::array<double, 6> const& getTXobs()    const { return tXobs;    }
+    std::array<double, 6> const& getTXobsErr() const { return tXobsErr; }
+
+    std::array<double, 6> const& getYobs()     const { return Yobs;     }
+    std::array<double, 6> const& getYobsErr()  const { return YobsErr;  }
+    std::array<double, 6> const& getTYobs()    const { return tYobs;    }
+    std::array<double, 6> const& getTYobsErr() const { return tYobsErr; }
+
+    std::array<double, 6> const& getZobs()     const { return Zobs;     }
+    std::array<double, 6> const& getZobsErr()  const { return ZobsErr;  }
+    std::array<double, 6> const& getTZobs()    const { return tZobs;    }
+    std::array<double, 6> const& getTZobsErr() const { return tZobsErr; }
+
   //========================= PRIVATE METHODS ==================================
   private: //===================================================================
 
@@ -45,9 +60,32 @@ class MillePedeFileReader {
     double Cutoffs[6] = {  Xcut_,  Ycut_,  Zcut_,
                           tXcut_, tYcut_, tZcut_};
 
+    bool PedeSuccess = false;
+    bool Movements   = false;
+    bool Error       = false;
+    bool Significant = false;
     bool updateDB    = false;
+    bool HitMax      = false;
+    bool HitErrorMax = false;
 
     int Nrec = 0;
+
+
+
+    std::array<double, 6> Xobs     = {{0.,0.,0.,0.,0.,0.}};
+    std::array<double, 6> XobsErr  = {{0.,0.,0.,0.,0.,0.}};
+    std::array<double, 6> tXobs    = {{0.,0.,0.,0.,0.,0.}};
+    std::array<double, 6> tXobsErr = {{0.,0.,0.,0.,0.,0.}};
+
+    std::array<double, 6> Yobs     = {{0.,0.,0.,0.,0.,0.}};
+    std::array<double, 6> YobsErr  = {{0.,0.,0.,0.,0.,0.}};
+    std::array<double, 6> tYobs    = {{0.,0.,0.,0.,0.,0.}};
+    std::array<double, 6> tYobsErr = {{0.,0.,0.,0.,0.,0.}};
+
+    std::array<double, 6> Zobs     = {{0.,0.,0.,0.,0.,0.}};
+    std::array<double, 6> ZobsErr  = {{0.,0.,0.,0.,0.,0.}};
+    std::array<double, 6> tZobs    = {{0.,0.,0.,0.,0.,0.}};
+    std::array<double, 6> tZobsErr = {{0.,0.,0.,0.,0.,0.}};
 
 };
 

--- a/Alignment/MillePedeAlignmentAlgorithm/interface/MillePedeFileReader.h
+++ b/Alignment/MillePedeAlignmentAlgorithm/interface/MillePedeFileReader.h
@@ -60,17 +60,8 @@ class MillePedeFileReader {
     double Cutoffs[6] = {  Xcut_,  Ycut_,  Zcut_,
                           tXcut_, tYcut_, tZcut_};
 
-    bool PedeSuccess = false;
-    bool Movements   = false;
-    bool Error       = false;
-    bool Significant = false;
     bool updateDB    = false;
-    bool HitMax      = false;
-    bool HitErrorMax = false;
-
     int Nrec = 0;
-
-
 
     std::array<double, 6> Xobs     = {{0.,0.,0.,0.,0.,0.}};
     std::array<double, 6> XobsErr  = {{0.,0.,0.,0.,0.,0.}};

--- a/Alignment/MillePedeAlignmentAlgorithm/src/MillePedeFileReader.cc
+++ b/Alignment/MillePedeAlignmentAlgorithm/src/MillePedeFileReader.cc
@@ -8,6 +8,8 @@
 /*** core framework functionality ***/
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
 
+
+
 //=============================================================================
 //===   PUBLIC METHOD IMPLEMENTATION                                        ===
 //=============================================================================
@@ -65,7 +67,10 @@ void MillePedeFileReader
         iss >> trash >> trash >> Nrec;
 
         if (Nrec < 25000) {
-
+          PedeSuccess = false;
+          Movements   = false;
+          Error       = false;
+          Significant = false;
           updateDB   = false;
         }
       }
@@ -74,6 +79,10 @@ void MillePedeFileReader
   } else {
     edm::LogError("MillePedeFileReader") << "Could not read millepede log-file.";
 
+    PedeSuccess = false;
+    Movements   = false;
+    Error       = false;
+    Significant = false;
     updateDB   = false;
     Nrec = 0;
   }
@@ -103,6 +112,7 @@ void MillePedeFileReader
       }
 
       if (tokens.size() > 4 /*3*/) {
+        PedeSuccess = true;
 
         int alignable      = std::stoi(tokens[0]);
         int alignableIndex = alignable % 10 - 1;
@@ -110,17 +120,66 @@ void MillePedeFileReader
         double ObsMove = std::stof(tokens[3]) * Multiplier[alignableIndex];
         double ObsErr  = std::stof(tokens[4]) * Multiplier[alignableIndex];
 
+        int det = -1;
+
+        if (alignable >= 60 && alignable <= 69) {
+          det = 2; // TPBHalfBarrel (x+)
+        } else if (alignable >= 8780 && alignable <= 8789) {
+          det = 3; // TPBHalfBarrel (x-)
+        } else if (alignable >= 17520 && alignable <= 17529) {
+          det = 4; // TPEHalfCylinder (x+,z+)
+        } else if (alignable >= 22380 && alignable <= 22389) {
+          det = 5; // TPEHalfCylinder (x-,z+)
+        } else if (alignable >= 27260 && alignable <= 27269) {
+          det = 0; // TPEHalfCylinder (x+,z-)
+        } else if (alignable >= 32120 && alignable <= 32129) {
+          det = 1; //TPEHalfCylinder (x-,z-)
+        } else {
+          continue;
+        }
+
+        if (alignableIndex == 0 && det >= 0 && det <= 5) {
+          Xobs[det] = ObsMove;
+          XobsErr[det] = ObsErr;
+        } else if (alignableIndex == 1 && det >= 0 && det <= 5) {
+          Yobs[det] = ObsMove;
+          YobsErr[det] = ObsErr;
+        } else if (alignableIndex == 2 && det >= 0 && det <= 5) {
+          Zobs[det] = ObsMove;
+          ZobsErr[det] = ObsErr;
+        } else if (alignableIndex == 3 && det >= 0 && det <= 5) {
+          tXobs[det] = ObsMove;
+          tXobsErr[det] = ObsErr;
+        } else if (alignableIndex == 4 && det >= 0 && det <= 5) {
+          tYobs[det] = ObsMove;
+          tYobsErr[det] = ObsErr;
+        } else if (alignableIndex == 5 && det >= 0 && det <= 5) {
+          tZobs[det] = ObsMove;
+          tZobsErr[det] = ObsErr;
+        }
+
 	if (std::abs(ObsMove) > maxMoveCut_) {
+          Movements   = false;
+          Error       = false;
+          Significant = false;
           updateDB    = false;
+          HitMax      = false;
           break;
 
         } else if (std::abs(ObsMove) > Cutoffs[alignableIndex]) {
 
+          Movements = true;
 	  if (std::abs(ObsErr) > maxErrorCut_) {
+            Error       = false;
+            Significant = false;
             updateDB    = false;
+            HitErrorMax = true;
             break;
           } else {
-	    if (std::abs(ObsMove/ObsErr) < sigCut_) {
+            Error = true;
+	    if (std::abs(ObsMove/ObsErr) > sigCut_) {
+              Significant = true;
+            } else {
 	      continue;
 	    } 
           }
@@ -131,6 +190,10 @@ void MillePedeFileReader
   } else {
     edm::LogError("MillePedeFileReader") << "Could not read millepede result-file.";
 
+    PedeSuccess = false;
+    Movements   = false;
+    Error       = false;
+    Significant = false;
     updateDB   = false;
     Nrec = 0;
   }

--- a/Alignment/MillePedeAlignmentAlgorithm/src/MillePedeFileReader.cc
+++ b/Alignment/MillePedeAlignmentAlgorithm/src/MillePedeFileReader.cc
@@ -67,10 +67,6 @@ void MillePedeFileReader
         iss >> trash >> trash >> Nrec;
 
         if (Nrec < 25000) {
-          PedeSuccess = false;
-          Movements   = false;
-          Error       = false;
-          Significant = false;
           updateDB   = false;
         }
       }
@@ -79,10 +75,6 @@ void MillePedeFileReader
   } else {
     edm::LogError("MillePedeFileReader") << "Could not read millepede log-file.";
 
-    PedeSuccess = false;
-    Movements   = false;
-    Error       = false;
-    Significant = false;
     updateDB   = false;
     Nrec = 0;
   }
@@ -112,7 +104,6 @@ void MillePedeFileReader
       }
 
       if (tokens.size() > 4 /*3*/) {
-        PedeSuccess = true;
 
         int alignable      = std::stoi(tokens[0]);
         int alignableIndex = alignable % 10 - 1;
@@ -159,29 +150,18 @@ void MillePedeFileReader
         }
 
 	if (std::abs(ObsMove) > maxMoveCut_) {
-          Movements   = false;
-          Error       = false;
-          Significant = false;
           updateDB    = false;
-          HitMax      = false;
           break;
 
         } else if (std::abs(ObsMove) > Cutoffs[alignableIndex]) {
-
-          Movements = true;
+	  
 	  if (std::abs(ObsErr) > maxErrorCut_) {
-            Error       = false;
-            Significant = false;
             updateDB    = false;
-            HitErrorMax = true;
             break;
           } else {
-            Error = true;
-	    if (std::abs(ObsMove/ObsErr) > sigCut_) {
-              Significant = true;
-            } else {
+  	    if (std::abs(ObsMove/ObsErr) < sigCut_) {
 	      continue;
-	    } 
+            } 
           }
 	  updateDB = true;
         }
@@ -190,10 +170,6 @@ void MillePedeFileReader
   } else {
     edm::LogError("MillePedeFileReader") << "Could not read millepede result-file.";
 
-    PedeSuccess = false;
-    Movements   = false;
-    Error       = false;
-    Significant = false;
     updateDB   = false;
     Nrec = 0;
   }

